cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/AbstractProgramRuntimeService.java

```diff
@@ -146,14 +145,13 @@ public final RuntimeInfo run(ProgramDescriptor programDescriptor, ProgramOptions
       Program executableProgram = createProgram(cConf, runner, programDescriptor, artifactDetail, tempDir);
       cleanUpTask = createCleanupTask(cleanUpTask, executableProgram);
 
-
       RuntimeInfo runtimeInfo = createRuntimeInfo(runner.run(executableProgram, optionsWithPlugins), programId,
-                                                  cleanUpTask);
+                                                  cleanUpTask, runtimeProgramOptions);
       monitorProgram(runtimeInfo, cleanUpTask);
       return runtimeInfo;
     } catch (Exception e) {
       // Set the program state to an error when an exception is thrown
-      programStateWriter.error(programId.run(runId), e);
+      programStateWriter.error(programId.run(runId), e, runtimeProgramOptions);
```

Comment:
you mentioned this is one of two places an error is written. Please remove the duplicates. I don't think this is the right place to write the error or to error log, as the caller needs to handle the exception anyway, and the logical thing for the caller to do is write an error and log an error.

Reply:
yes, i will change the logic to throw the error.